src/test/java/com/github/checkstyle/regression/internal/GitUtils.java

```diff
@@ -63,6 +67,7 @@ public static File addAnEmptyFileAndCommit(Repository repository, String fileNam
             throws IOException, GitAPIException {
         try (Git git = new Git(repository)) {
             final File file = new File(repository.getDirectory().getParent(), fileName);
+            file.getParentFile().mkdirs();
```

Comment:
I see now, we are passing a string to create a file in a specific path.
This is ok.